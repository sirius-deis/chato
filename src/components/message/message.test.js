import { render, screen } from "@testing-library/react";
import Message from "./message";

const time = new Date("2024-07-17T21:08:30").toISOString();

const message = {
  message: "Test Message",
  createdAt: time,
};

describe("Message component", () => {
  it("should match snapshot", () => {
    const { container } = render(<Message>{message}</Message>);
    expect(container).toMatchSnapshot();
  });
  it("should contain provided text", () => {
    render(<Message>{message}</Message>);
    expect(screen.getByText(/Test Message/)).toBeInTheDocument();
  });
  it("should have a timestamp", () => {
    render(<Message>{message}</Message>);
    expect(screen.getByText(/[0-9]{2}:[0-9]{2}:[0-9]{2}/)).toBeInTheDocument();
  });
});
