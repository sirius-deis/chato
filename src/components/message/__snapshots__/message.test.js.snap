// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should match snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg cZkKqC  "
  >
    Test Message
    <div
      class="sc-gEvEer jozdBC"
    >
      19:08:30
    </div>
  </div>
</div>
`;
